### ISSUE TITLE
Cleanup list.arr's duplicate functions

### ISSUE THREAD

**Issue by blerner** [brownplt/pyret-lang#33]
`map` is both a method on lists and a free-floating function, and the mapping algorithm is written twice.  Ditto `each`.  But there is no `list.filter` corresponding to the `filter` method...

I think since `map2`, `each3`, etc exist, and corresponding `filter#` might be useful, the methods should possibly delegate to the functions.  For efficiency (hah! ;-)) you may not want to do this, but it seems conceptually clearer...

**Comment by Emmay** [comment]
Is this still an issue?

**Comment by blerner** [comment]
Yes, sadly.  We still haven't done a thorough pass over what names we want in the global scope as functions, in the module scope as functions, only available as methods, and what the naming conventions are...
